Examples/CliClient/Program.cs

```diff
@@ -340,30 +370,23 @@ async Task<Connection> CreateUpgradeConnection(string host, int port)
             .SetHttp2Settings(config.Settings)
             .Build();
 
-        // Create a TCP connection
-        logger.LogInformation($"Starting to connect to {host}:{port}");
-        var tcpClient = new TcpClient();
-        await tcpClient.ConnectAsync(host, port);
-        tcpClient.Client.NoDelay = true;
-        logger.LogInformation("Connected to remote");
-
-        // Create HTTP/2 stream abstraction on top of the socket
-        var wrappedStreams = tcpClient.Client.CreateStreams();
-        var upgradeReadStream = new UpgradeReadStream(wrappedStreams.ReadableStream);
+        var (readableStream, writeableStream) = await CreateStreams(scheme, host, port);
```

Comment:
If I understand the specification correctly then HTTP/2 via upgrade is only supported for not encrypted connections (h2c). So while this might work it seems to be against the spec.
Maybe just throw an exception that upgrade+TLS is not supported instead.

If this would work, then the ALPN modifier should also be not set - since the upgrade intent is signaled via HTTP upgrade instead of via ALPN for this case.

Reply:
Ah yes, I will throw a `NotSupportException` in this case.